plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -52,15 +52,16 @@
         executor.execute(format("ALTER DATABASE %s SET READ_COMMITTED_SNAPSHOT ON", databaseName));
     };
 
-    private static final RetryPolicy<InitializedState> CONTAINER_RETRY_POLICY = new RetryPolicy<InitializedState>()
+    private static final RetryPolicy<InitializedState> CONTAINER_RETRY_POLICY = RetryPolicy.<InitializedState>builder()
             .withBackoff(1, 5, ChronoUnit.SECONDS)
             .withMaxRetries(5)
             .handleIf(throwable -> getCausalChain(throwable).stream()
-                    .anyMatch(SQLException.class::isInstance))
+                    .anyMatch(SQLException.class::isInstance) || throwable.getMessage().contains("Container exited with code"))
```

Comment:
This can also lead to retries which never succeed. Do we know why container exits?

Reply:
Yes. Deadlock when trying to set collation. I believe it was a new flaky - that's why I've added this one